lib/spack/spack/cmd/add.py

```diff
@@ -17,8 +16,7 @@
 
 
 def setup_parser(subparser):
-    subparser.add_argument(
-        'specs', nargs=argparse.REMAINDER, help="specs of packages to add")
+    arguments.add_common_arguments(subparser, ['specs'])
```

Comment:
@opadron Here is an example of how a command would make use of the common argument for specs so that either clispecs or yaml paths are accepted.